Additional fixes for PR 862

Onepamopa opened the pull request:
I couldn't commit directly to the PR, doing a new one with all (862 + my fixes).

Onepamopa committed:
Additional fixes for PR 862

I couldn't commit directly to the PR, doing a new one with all (862 + my fixes).

Onepamopa commented:
A few more fixes incoming...

Onepamopa committed:
Blueprints fix

Onepamopa commented:
@dwainscheeren how to define the sensors for temperature/humidity manually for an area?
I can't find any references in any of the python files?!

Ideally I'd like to be able to define these via edit mode on areas. Especially in cases where 2 devices provide temperature sensors in the same area, it averages out (not something I'd like), or, the area doesn't even "see" the temperature/humidity sensor (when provided by a climate entity)

dwainscheeren commented:
Someone tested this pr?

Onepamopa commented:
> Someone tested this pr?

Besides myself and one of my friends? :) Who knows...

ForumHulp commented:
I tested it and was working ok, but with the latest ha update it stopped at loading pythonsafeloader in process_yaml

tristanXme commented:
/edited

See here:

https://github.com/dwainscheeren/dwains-lovelace-dashboard/pull/863#issuecomment-2775410415

ForumHulp commented:
ok working for me

Onepamopa commented:
I haven't yet upgraded to the latest HA version.
@tristanXme I'll check your changes against mine and apply them in this PR.

de-Techneut commented:
Can confirm the update of @tristanXme is working. Can see dwains dashboard again. Can also confirm this breaks the "More Page".

Onepamopa commented:
I'll try to preserve more_pages ... hitting a wall atm, but we'll see.
Looks like it's related to:
https://github.com/home-assistant/core/pull/140683/commits/1811733fd3ca55e013e49068e6ad3b105118b769

Onepamopa commented:
I have some progress.... doing a last reboot to ensure it works.

Onepamopa committed:
Fix for the issues caused by the last HASS update.

Kindly test and let me know if all is good.

Onepamopa commented:
Step on it :)

tristanXme commented:
Thank you so much — I can confirm everything is working!

For those who applied my temporary fix, or for those who don't know how to change the files manually, I have attached a ZIP file. Just overwrite your files (always make a backup first) and restart Home Assistant. All changes from Onepamopa are included, and my changes have been reverted.

After that, everything should be fully working again.

[custom_components.zip](https://github.com/user-attachments/files/19585692/custom_components.zip)

See https://github.com/dwainscheeren/dwains-lovelace-dashboard/pull/863#issuecomment-2776415183 for latest ZIP File

znanywawa commented:
> Thank you so much — I can confirm everything is working!
> 
> For those who applied my temporary fix, or for those who don't know how to change the files manually, I have attached a ZIP file. Just overwrite your files (always make a backup first) and restart Home Assistant. All changes from Onepamopa are included, and my changes have been reverted.
> 
> After that, everything should be fully working again.
> 
> [custom_components.zip](https://github.com/user-attachments/files/19585692/custom_components.zip)

I have used Your zip file at and replaced only files from it to my installation, unfortunately at HA 2025.3.4 and got following error:
```
Logger: homeassistant.setup
Source: setup.py:340
First occurred: 14:27:02 (1 occurrences)
Last logged: 14:27:02

Setup failed for custom integration 'dwains_dashboard': Unable to import component: No module named 'annotatedyaml'
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/setup.py", line 340, in _async_setup_component
    component = await integration.async_get_component()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1034, in async_get_component
    self._component_future.result()
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1014, in async_get_component
    comp = await self.hass.async_add_import_executor_job(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        self._get_component, True
        ^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/usr/local/lib/python3.13/concurrent/futures/thread.py", line 59, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1074, in _get_component
    ComponentProtocol, importlib.import_module(self.pkg_path)
                       ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/util/loop.py", line 201, in protected_loop_func
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.13/importlib/__init__.py", line 88, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 1026, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/config/custom_components/dwains_dashboard/__init__.py", line 12, in <module>
    from .process_yaml import process_yaml, reload_configuration
  File "/config/custom_components/dwains_dashboard/process_yaml.py", line 16, in <module>
    from annotatedyaml import loader
ModuleNotFoundError: No module named 'annotatedyaml'
```

tristanXme commented:
> > Thank you so much — I can confirm everything is working!
> > For those who applied my temporary fix, or for those who don't know how to change the files manually, I have attached a ZIP file. Just overwrite your files (always make a backup first) and restart Home Assistant. All changes from Onepamopa are included, and my changes have been reverted.
> > After that, everything should be fully working again.
> > [custom_components.zip](https://github.com/user-attachments/files/19585692/custom_components.zip)
> 
> I have used Your zip file at and replaced only files from it to my installation, unfortunately at HA 2025.3.4 and got following error:
> 
> ```
> Logger: homeassistant.setup
> Source: setup.py:340
> First occurred: 14:27:02 (1 occurrences)
> Last logged: 14:27:02
> 
> Setup failed for custom integration 'dwains_dashboard': Unable to import component: No module named 'annotatedyaml'
> Traceback (most recent call last):
>   File "/usr/src/homeassistant/homeassistant/setup.py", line 340, in _async_setup_component
>     component = await integration.async_get_component()
>                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/src/homeassistant/homeassistant/loader.py", line 1034, in async_get_component
>     self._component_future.result()
>     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^
>   File "/usr/src/homeassistant/homeassistant/loader.py", line 1014, in async_get_component
>     comp = await self.hass.async_add_import_executor_job(
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>         self._get_component, True
>         ^^^^^^^^^^^^^^^^^^^^^^^^^
>     )
>     ^
>   File "/usr/local/lib/python3.13/concurrent/futures/thread.py", line 59, in run
>     result = self.fn(*self.args, **self.kwargs)
>   File "/usr/src/homeassistant/homeassistant/loader.py", line 1074, in _get_component
>     ComponentProtocol, importlib.import_module(self.pkg_path)
>                        ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
>   File "/usr/src/homeassistant/homeassistant/util/loop.py", line 201, in protected_loop_func
>     return func(*args, **kwargs)
>   File "/usr/local/lib/python3.13/importlib/__init__.py", line 88, in import_module
>     return _bootstrap._gcd_import(name[level:], package, level)
>            ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
>   File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
>   File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
>   File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
>   File "<frozen importlib._bootstrap_external>", line 1026, in exec_module
>   File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
>   File "/config/custom_components/dwains_dashboard/__init__.py", line 12, in <module>
>     from .process_yaml import process_yaml, reload_configuration
>   File "/config/custom_components/dwains_dashboard/process_yaml.py", line 16, in <module>
>     from annotatedyaml import loader
> ModuleNotFoundError: No module named 'annotatedyaml'
> ```

Just double checked my archiv - its correct. You have issues with changes that Onepamopa made in the process_yaml.py

Are you sure you are on the lastest HA (Core) Version?


Core
2025.4.0
Supervisor
2025.03.4
Operating System
15.1
Frontend
20250401.0

Working for me

znanywawa commented:
> > > Thank you so much — I can confirm everything is working!
> > > For those who applied my temporary fix, or for those who don't know how to change the files manually, I have attached a ZIP file. Just overwrite your files (always make a backup first) and restart Home Assistant. All changes from Onepamopa are included, and my changes have been reverted.
> > > After that, everything should be fully working again.
> > > [custom_components.zip](https://github.com/user-attachments/files/19585692/custom_components.zip)
> > 
> > 
> > I have used Your zip file at and replaced only files from it to my installation, unfortunately at HA 2025.3.4 and got following error:
> > ```
> > Logger: homeassistant.setup
> > Source: setup.py:340
> > First occurred: 14:27:02 (1 occurrences)
> > Last logged: 14:27:02
> > 
> > Setup failed for custom integration 'dwains_dashboard': Unable to import component: No module named 'annotatedyaml'
> > Traceback (most recent call last):
> >   File "/usr/src/homeassistant/homeassistant/setup.py", line 340, in _async_setup_component
> >     component = await integration.async_get_component()
> >                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> >   File "/usr/src/homeassistant/homeassistant/loader.py", line 1034, in async_get_component
> >     self._component_future.result()
> >     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^
> >   File "/usr/src/homeassistant/homeassistant/loader.py", line 1014, in async_get_component
> >     comp = await self.hass.async_add_import_executor_job(
> >            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> >         self._get_component, True
> >         ^^^^^^^^^^^^^^^^^^^^^^^^^
> >     )
> >     ^
> >   File "/usr/local/lib/python3.13/concurrent/futures/thread.py", line 59, in run
> >     result = self.fn(*self.args, **self.kwargs)
> >   File "/usr/src/homeassistant/homeassistant/loader.py", line 1074, in _get_component
> >     ComponentProtocol, importlib.import_module(self.pkg_path)
> >                        ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
> >   File "/usr/src/homeassistant/homeassistant/util/loop.py", line 201, in protected_loop_func
> >     return func(*args, **kwargs)
> >   File "/usr/local/lib/python3.13/importlib/__init__.py", line 88, in import_module
> >     return _bootstrap._gcd_import(name[level:], package, level)
> >            ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> >   File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
> >   File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
> >   File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
> >   File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
> >   File "<frozen importlib._bootstrap_external>", line 1026, in exec_module
> >   File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
> >   File "/config/custom_components/dwains_dashboard/__init__.py", line 12, in <module>
> >     from .process_yaml import process_yaml, reload_configuration
> >   File "/config/custom_components/dwains_dashboard/process_yaml.py", line 16, in <module>
> >     from annotatedyaml import loader
> > ModuleNotFoundError: No module named 'annotatedyaml'
> > ```
> 
> Just double checked my archiv - its correct. You have issues with changes that Onepamopa made in the process_yaml.py
> 
> Are you sure you are on the lastest HA (Core) Version?
> 
> Core 2025.4.0 Supervisor 2025.03.4 Operating System 15.1 Frontend 20250401.0
> 
> Working for me

No, I tried it on Core 2025.3.4 to check it before updating to 2025.4.0
Exact details:
Core: 2025.3.4
Supervisor: 2025.03.4
Operating System: 15.1
Frontend: 20250306.0

So as I understand, its only compatible with 2025.4.0 ?

tristanXme commented:
Yes, I think it only works with the newest Version.

znanywawa commented:
> Yes, I think it only works with the newest Version.

Mkay, thanks!
Im gonna test it and give feedback :)
@tristanXme, seems that works after update to 2025.4.0.
Great job guys!

Ive noticed that its making a lot of noise in Logs, is same for You?

```
Logger: custom_components.dwains_dashboard.process_yaml
Source: custom_components/dwains_dashboard/process_yaml.py:144
integration: Dwains Dashboard ([documentation](https://dwainscheeren.github.io/dwains-lovelace-dashboard/), [issues](https://github.com/dwainscheeren/dwains-lovelace-dashboard/issues))
First occurred: 15:05:06 (4 occurrences)
Last logged: 15:05:06

process_yaml() config.yaml exists, system
process_yaml() config.yaml exists, media
process_yaml() config.yaml exists, vacuum
process_yaml() config.yaml exists, batteries
```

```
Logger: homeassistant.util.logging
Source: util/logging.py:74
First occurred: 15:06:22 (1 occurrences)
Last logged: 15:06:22

Module custom_components.dwains_dashboard is logging too frequently. 200 messages since last count
```

```
Logger: custom_components.dwains_dashboard
Source: custom_components/dwains_dashboard/__init__.py:181
integration: Dwains Dashboard ([documentation](https://dwainscheeren.github.io/dwains-lovelace-dashboard/), [issues](https://github.com/dwainscheeren/dwains-lovelace-dashboard/issues))
First occurred: 15:05:57 (888 occurrences)
Last logged: 15:12:56

area_cards: conditional20240117141414.yaml
area_cards: conditional20240117141632.yaml
area_cards: conditional20240119123308.yaml
area_cards: conditional20240119123423.yaml
area_cards: conditional20240119123527.yaml
```

```
Logger: custom_components.dwains_dashboard
Source: custom_components/dwains_dashboard/__init__.py:212
integration: Dwains Dashboard ([documentation](https://dwainscheeren.github.io/dwains-lovelace-dashboard/), [issues](https://github.com/dwainscheeren/dwains-lovelace-dashboard/issues))
First occurred: 15:06:00 (518 occurrences)
Last logged: 15:12:56

entity_cards: light.kid_room_bedside_lamp.yaml
entity_cards: media_player.living_room_tv.yaml
entity_cards: light.corner_bed_leds.yaml
entity_cards: media_player.living_room_soundbar_ms750.yaml
entity_cards: person.sunny.yaml
```

```
Logger: custom_components.dwains_dashboard
Source: custom_components/dwains_dashboard/__init__.py:226
integration: Dwains Dashboard ([documentation](https://dwainscheeren.github.io/dwains-lovelace-dashboard/), [issues](https://github.com/dwainscheeren/dwains-lovelace-dashboard/issues))
First occurred: 15:06:01 (555 occurrences)
Last logged: 15:12:56

entities_popup: person.znany.yaml
entities_popup: binary_sensor.balcony_doors_contact.yaml
entities_popup: binary_sensor.bedroom_window_contact.yaml
entities_popup: binary_sensor.living_room_window_contact_2.yaml
entities_popup: person.sunny.yaml
```

```
Logger: custom_components.dwains_dashboard
Source: custom_components/dwains_dashboard/__init__.py:241
integration: Dwains Dashboard ([documentation](https://dwainscheeren.github.io/dwains-lovelace-dashboard/), [issues](https://github.com/dwainscheeren/dwains-lovelace-dashboard/issues))
First occurred: 15:06:03 (74 occurrences)
Last logged: 15:12:56

devices_card: light.yaml
```

Onepamopa committed:
HA version check

I don't know if this minimum version check would work, maybe Dwain would have more info.

Onepamopa commented:
@znanywawa the log noise was for debugging purposes. Feel free to comment them out.

znanywawa commented:
> @znanywawa the log noise was for debugging purposes. Feel free to comment them out.

Thanks!
Commented and fingers crossed for official release :)
Thanks guys for fix!

Onepamopa commented:
@dwainscheeren small issue - noticed when you add / remove an Area icon - it doesn't get reflected immediately. Only after HA is reloaded does the change show up.

Which function is related to it?

bennitschke commented:
Thank you so much for the fix!!

bangerdemon commented:
> Thank you so much — I can confirm everything is working!
> 
> For those who applied my temporary fix, or for those who don't know how to change the files manually, I have attached a ZIP file. Just overwrite your files (always make a backup first) and restart Home Assistant. All changes from Onepamopa are included, and my changes have been reverted.
> 
> After that, everything should be fully working again.
> 
> [custom_components.zip](https://github.com/user-attachments/files/19585692/custom_components.zip)

Can you give me a walkthrough of what needs changing please? Not tech savvy, and don't want to completely kill my HA!

tristanXme commented:
> > Thank you so much — I can confirm everything is working!
> > For those who applied my temporary fix, or for those who don't know how to change the files manually, I have attached a ZIP file. Just overwrite your files (always make a backup first) and restart Home Assistant. All changes from Onepamopa are included, and my changes have been reverted.
> > After that, everything should be fully working again.
> > [custom_components.zip](https://github.com/user-attachments/files/19585692/custom_components.zip)
> 
> Can you give me a walkthrough of what needs changing please? Not tech savvy, and don't want to completely kill my HA!

You can see the changed code here:
https://github.com/dwainscheeren/dwains-lovelace-dashboard/pull/863/files

tristanXme commented:
> Ive noticed that its making a lot of noise in Logs, is same for You?

Yes. I also commented it out for me, but I did not want to include it in the files I uploaded.

w1Ngx commented:
Hm, i uploaded the folder provided and Dwains Dashboard comes up again. However, it shows the default fresh dashboard including all entities everywhere and the navigation bar is not working. Cache already emptied and reloaded, no error or other messages in the logs.

Onepamopa commented:
> Hm, i uploaded the folder provided and Dwains Dashboard comes up again. However, it shows the default fresh dashboard including all entities everywhere and the navigation bar is not working. Cache already emptied and reloaded, no error or other messages in the logs.

Where did you upload it, exactly? Because I haven't had an issue with my dashboard - it's working normally.

w1Ngx commented:
I uploaded it to the config/custom_components/dwains_dashboard folder

I attached a screenshot, also with a lot of errors in the 04.more_page_view.yaml

![image](https://github.com/user-attachments/assets/b21bcb26-76ed-431c-917c-f2403a77cf49)

Onepamopa commented:
These aren't errors. Make sure you replaced the files from the .zip archive into the correct folder.
After that - restart HA.

bangerdemon commented:
> > > Thank you so much — I can confirm everything is working!
> > > For those who applied my temporary fix, or for those who don't know how to change the files manually, I have attached a ZIP file. Just overwrite your files (always make a backup first) and restart Home Assistant. All changes from Onepamopa are included, and my changes have been reverted.
> > > After that, everything should be fully working again.
> > > [custom_components.zip](https://github.com/user-attachments/files/19585692/custom_components.zip)
> > 
> > 
> > Can you give me a walkthrough of what needs changing please? Not tech savvy, and don't want to completely kill my HA!
> 
> You can see the changed code here: https://github.com/dwainscheeren/dwains-lovelace-dashboard/pull/863/files

Wow!!! I'll look for another dashboard I think!

Cheers anyway.

Onepamopa commented:
> > > > Thank you so much — I can confirm everything is working!
> > > > For those who applied my temporary fix, or for those who don't know how to change the files manually, I have attached a ZIP file. Just overwrite your files (always make a backup first) and restart Home Assistant. All changes from Onepamopa are included, and my changes have been reverted.
> > > > After that, everything should be fully working again.
> > > > [custom_components.zip](https://github.com/user-attachments/files/19585692/custom_components.zip)
> > > 
> > > 
> > > Can you give me a walkthrough of what needs changing please? Not tech savvy, and don't want to completely kill my HA!
> > 
> > 
> > You can see the changed code here: https://github.com/dwainscheeren/dwains-lovelace-dashboard/pull/863/files
> 
> Wow!!! I'll look for another dashboard I think!
> 
> Cheers anyway.

Jesus.. not everything can be presented on a silver platter. Some understanding is naturally required, if you're not up to it - maybe you should move away from home assistant alltogether.

tristanXme commented:
> > > > > Thank you so much — I can confirm everything is working!
> > > > > For those who applied my temporary fix, or for those who don't know how to change the files manually, I have attached a ZIP file. Just overwrite your files (always make a backup first) and restart Home Assistant. All changes from Onepamopa are included, and my changes have been reverted.
> > > > > After that, everything should be fully working again.
> > > > > [custom_components.zip](https://github.com/user-attachments/files/19585692/custom_components.zip)
> > > > 
> > > > 
> > > > Can you give me a walkthrough of what needs changing please? Not tech savvy, and don't want to completely kill my HA!
> > > 
> > > 
> > > You can see the changed code here: https://github.com/dwainscheeren/dwains-lovelace-dashboard/pull/863/files
> > 
> > 
> > Wow!!! I'll look for another dashboard I think!
> > Cheers anyway.
> 
> Jesus.. not everything can be presented on a silver platter. Some understanding is naturally required, if you're not up to it - maybe you should move away from home assistant alltogether.

If u dont know how to do it, just use my zip file and overwrite the existing files & restart HA

w1Ngx commented:
good luck, i didn’t find any with these features, but only some that take ages to configure ;-)

Onepamopa commented:
@bangerdemon if you want everything "ready" - you should wait for @dwainscheeren to approve the pull request into a new "official" release. But I bet he'd have a few other things to fix before doing so.

w1Ngx commented:
> > > > > > Thank you so much — I can confirm everything is working!
> > > > > > For those who applied my temporary fix, or for those who don't know how to change the files manually, I have attached a ZIP file. Just overwrite your files (always make a backup first) and restart Home Assistant. All changes from Onepamopa are included, and my changes have been reverted.
> > > > > > After that, everything should be fully working again.
> > > > > > [custom_components.zip](https://github.com/user-attachments/files/19585692/custom_components.zip)
> > > > > 
> > > > > 
> > > > > Can you give me a walkthrough of what needs changing please? Not tech savvy, and don't want to completely kill my HA!
> > > > 
> > > > 
> > > > You can see the changed code here: https://github.com/dwainscheeren/dwains-lovelace-dashboard/pull/863/files
> > > 
> > > 
> > > Wow!!! I'll look for another dashboard I think!
> > > Cheers anyway.
> > 
> > 
> > Jesus.. not everything can be presented on a silver platter. Some understanding is naturally required, if you're not up to it - maybe you should move away from home assistant alltogether.
> 
> If u dont know how to do it, just use my zip file and overwrite the existing files & restart HA

wish that worked for me. I am trying to find an indicator what’s not working, but there is nothing somehow

dwainscheeren commented:
> @bangerdemon if you want everything "ready" - you should wait for @dwainscheeren to approve the pull request into a new "official" release. But I bet he'd have a few other things to fix before doing so.

If I know for 100% this PR works then I can merge it in an official release this weekend :)

Xander-V commented:
Can confirm dashboard is loading again for me as well after the upgraded files in the zip. Using 2025.4.0 in docker.

More-pages functional as well, although I get the below error. Seems there's a double call somewhere?
(And an error/remark that you're logging too much now, but that can of course wait till the actual merge is done.)

```
Logger: custom_components.dwains_dashboard.process_yaml
Source: custom_components/dwains_dashboard/process_yaml.py:144
integration: Dwains Dashboard (documentation, issues)
First occurred: 16:20:21 (7 occurrences)
Last logged: 16:20:21

process_yaml() config.yaml exists, persons
process_yaml() config.yaml exists, benzineprijzen
process_yaml() config.yaml exists, afvalwijzer_page_blueprint
process_yaml() config.yaml exists, system_info_page
process_yaml() config.yaml exists, docker_page 
```

Onepamopa commented:
> Can confirm dashboard is loading again for me as well after the upgraded files in the zip. Using 2025.4.0 in docker.
> 
> More-pages functional as well, although I get the below error. Seems there's a double call somewhere? (And an error/remark that you're logging too much now, but that can of course wait till the actual merge is done.)
> 
> ```
> Logger: custom_components.dwains_dashboard.process_yaml
> Source: custom_components/dwains_dashboard/process_yaml.py:144
> integration: Dwains Dashboard (documentation, issues)
> First occurred: 16:20:21 (7 occurrences)
> Last logged: 16:20:21
> 
> process_yaml() config.yaml exists, persons
> process_yaml() config.yaml exists, benzineprijzen
> process_yaml() config.yaml exists, afvalwijzer_page_blueprint
> process_yaml() config.yaml exists, system_info_page
> process_yaml() config.yaml exists, docker_page 
> ```

That's not an error. I'll remove the debug logging.

Onepamopa committed:
Disable debug logging

Onepamopa commented:
Here's a new zip file up to the last commit, for whoever wants to test.

[dwains_dashboard.zip](https://github.com/user-attachments/files/19591604/dwains_dashboard.zip)

thebatfink commented:
Thanks for working on this. The files in the zip fixed the dashboard in latest core. Had me panicked.

edit: latest zip above also seems to work well.

thowiegit commented:
> Here's a new zip file up to the last commit, for whoever wants to test.

nice, that works like a charme!

tristanXme commented:
> Here's a new zip file up to the last commit, for whoever wants to test.
> 
> [dwains_dashboard.zip](https://github.com/user-attachments/files/19591604/dwains_dashboard.zip)

tested it - works :)

vulcanoid1975 commented:
Tested as well, It works. Thank You all for the effort in the delivery of this.

yurtalan92 commented:
I am not good at these situations, I am afraid that if I change the original files and have a problem, I will not be able to fix it.
So should I wait for an update without updating the zip file files? Will it be fixed with an update? I do not use it in any other dashboard. All my devices' cards were in dwainsdashboard :)

dfzamora commented:
Confirmed it work perfectly fine…. People make sure to take a backup just in case...

Atlas-Night-Out commented:
> > Here's a new zip file up to the last commit, for whoever wants to test.
> > [dwains_dashboard.zip](https://github.com/user-attachments/files/19591604/dwains_dashboard.zip)
> 
> tested it - works :)

How do i install this dwain as vanished since i undated HA core tonight and not sure what to do

mikkelke commented:
Thanks for the fix! Here is what to do:

🛠️ Fix for Dwains Dashboard on Home Assistant 2025.4

Home Assistant 2025.4 breaks Dwains Dashboard. There's a fix available via this pull request:
🔗 https://github.com/dwainscheeren/dwains-lovelace-dashboard/pull/863
✅ Quick Fix Instructions

    Download the fixed version from the PR:
    📦 https://github.com/dwainscheeren/dwains-lovelace-dashboard/files/14743618/dwains-dashboard-2025.4-fix.zip

    Unzip it on your computer.

    Upload the contents of the ZIP to this folder on your Home Assistant system:

    /homeassistant/custom_components/dwains_dashboard/

    Restart Home Assistant.

That’s it! The dashboard should now work correctly on Home Assistant 2025.4.

Once the fix is officially merged and released, you can switch back to the official version.

thebatfink commented:
> I am not good at these situations, I am afraid that if I change the original files and have a problem, I will not be able to fix it. So should I wait for an update without updating the zip file files? Will it be fixed with an update? I do not use it in any other dashboard. All my devices' cards were in dwainsdashboard :)

So make a backup of the Dwains folder before replacing the files? Does it really matter when the dashboard flat out doesn’t work, how much worse could you make it anyway?

Cook23 commented:
Works correctly thanks

mikkelke commented:
> > I am not good at these situations, I am afraid that if I change the original files and have a problem, I will not be able to fix it. So should I wait for an update without updating the zip file files? Will it be fixed with an update? I do not use it in any other dashboard. All my devices' cards were in dwainsdashboard :)
> 
> So make a backup of the Dwains folder before replacing the files? Does it really matter when the dashboard flat out doesn’t work, how much worse could you make it anyway?

Very valid point. I even tried rolling my docker installation back to 2025.3. But there is breaking changes related to the core functionality. Rolling back ended up the recorder and other core components not working. Dwains Dashboard worked again, but the old files have no use if you have upgraded.

Onepamopa commented:
> > > Here's a new zip file up to the last commit, for whoever wants to test.
> > > [dwains_dashboard.zip](https://github.com/user-attachments/files/19591604/dwains_dashboard.zip)
> > 
> > 
> > tested it - works :)
> 
> How do i install this dwain as vanished since i undated HA core tonight and not sure what to do

Just replace the files in config/custom_components/dwains_dashboard/ with the contents of the zip file, overwriting the old files. It'll start working.

Atlas-Night-Out commented:
How do i unload to /homeassistant/custom_components/dwains_dashboard/

Onepamopa commented:
> How do i unload to /homeassistant/custom_components/dwains_dashboard/

Use a file manager addon?

Atlas-Night-Out commented:
I know to use Studio Code Server but not seure how to add the files to it? or if i replace them?

albertolinard commented:
I'm glad it worked! I replaced the contents of my current directory with the contents of the zip file, and my old Homekit Infused 2022.02.0—which had stopped working because Dwains Dashboard was broken—started working again. I use Dwains on my phone and Homekit on my computer, and both were somehow connected. Thanks to everyone involved!

vulcanoid1975 commented:
> I know to use Studio Code Server but not sure how to add the files to it? or if i replace them?
> 
> ![Studio Code Server](https://private-user-images.githubusercontent.com/83882497/430303718-46b0cbf0-49ad-4b32-bbae-020f66b3ee6b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDM3NjI3ODQsIm5iZiI6MTc0Mzc2MjQ4NCwicGF0aCI6Ii84Mzg4MjQ5Ny80MzAzMDM3MTgtNDZiMGNiZjAtNDlhZC00YjMyLWJiYWUtMDIwZjY2YjNlZTZiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDA0VDEwMjgwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThkMTBiNDFmNjhhZGE0ZWQwZjgwMmU2NjBlMmI3MDIyOTYxZTYwNDc1MDhiY2FiYTYwZGVhNTBkOWZjMTRjMGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.gEkBjXD-OUppMDeske1MIot5GEYSCztaGt4XHdR0K3Y)

My suggestion is to use:
https://github.com/alexbelgium/hassio-addons

Add the Repository to the Home Assistant Add-ons (refresh page after adding) and then look for "Filebrowser", install and run.

It's like a Windows File Explorer for Home Assistant (use with caution of course).

Hope that helps ...

Atlas-Night-Out commented:
> > How do i unload to /homeassistant/custom_components/dwains_dashboard/
> 
> Use a file manager addon?

Install the FTP in Addons and   use a FTP program after you have configured the FTP in Home Assistant.

![ftp settings](https://github.com/user-attachments/assets/c4d6a07c-9439-498e-8483-a04fe5ebdedb)

See this video which is very helpful:


[Transfer files to Home Assistant with FTP](https://www.youtube.com/watch?v=U_h49OoFl-g&t=36s)

Atlas-Night-Out commented:
> > Here's a new zip file up to the last commit, for whoever wants to test.
> > [dwains_dashboard.zip](https://github.com/user-attachments/files/19591604/dwains_dashboard.zip)
> 
> tested it - works :)

Thank you so far all is back up and working with Dwain! thanks 

I did struggle to find the directory as there is 2 folder in there! So be careful where you upload too! But ts all working!

mikkelke commented:
> > > How do i unload to /homeassistant/custom_components/dwains_dashboard/
> > 
> > 
> > Use a file manager addon?
> 
> Install the FTP in Addons and use a FTP program after you have configured the FTP in Home Assistant.
> 
> ![ftp settings](https://private-user-images.githubusercontent.com/83882497/430377712-c4d6a07c-9439-498e-8483-a04fe5ebdedb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDM3NzQxNjYsIm5iZiI6MTc0Mzc3Mzg2NiwicGF0aCI6Ii84Mzg4MjQ5Ny80MzAzNzc3MTItYzRkNmEwN2MtOTQzOS00OThlLTg0ODMtYTA0ZmU1ZWJkZWRiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDA0VDEzMzc0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRhNzg4ZjZlYWFlZjhiOGEwYmUzYzVjNWZmMmJhMDc5MGViOTg1MjkyNjJiMTYwZjczMWJjMDlkMjQyZTMxYjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.w9HOa4dCbFfYB4lyhemovOhzRHUWltM6rp9TgCIpQI8)
> 
> See this video which is very helpful:
> 
> [Transfer files to Home Assistant with FTP](https://www.youtube.com/watch?v=U_h49OoFl-g&t=36s)

If you use filezilla you could also just use sftp and connect with ssh instead of setting up a FTP add-on that does the same.

Cook23 commented:
> > > > How do i unload to /homeassistant/custom_components/dwains_dashboard/
> > > 
> > > 
> > > Use a file manager addon?
> > 
> > 
> > Install the FTP in Addons and use a FTP program after you have configured the FTP in Home Assistant.
> > ![ftp settings](https://private-user-images.githubusercontent.com/83882497/430377712-c4d6a07c-9439-498e-8483-a04fe5ebdedb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDM3NzQxNjYsIm5iZiI6MTc0Mzc3Mzg2NiwicGF0aCI6Ii84Mzg4MjQ5Ny80MzAzNzc3MTItYzRkNmEwN2MtOTQzOS00OThlLTg0ODMtYTA0ZmU1ZWJkZWRiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDA0VDEzMzc0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRhNzg4ZjZlYWFlZjhiOGEwYmUzYzVjNWZmMmJhMDc5MGViOTg1MjkyNjJiMTYwZjczMWJjMDlkMjQyZTMxYjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.w9HOa4dCbFfYB4lyhemovOhzRHUWltM6rp9TgCIpQI8)
> > See this video which is very helpful:
> > [Transfer files to Home Assistant with FTP](https://www.youtube.com/watch?v=U_h49OoFl-g&t=36s)
> 
> If you use filezilla you could also just use sftp and connect with ssh instead of setting up a FTP add-on that does the same.

Or install samba addon

Passi66 commented:
> > I know to use Studio Code Server but not sure how to add the files to it? or if i replace them?
> > ![Studio Code Server](https://private-user-images.githubusercontent.com/83882497/430303718-46b0cbf0-49ad-4b32-bbae-020f66b3ee6b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDM3NjI3ODQsIm5iZiI6MTc0Mzc2MjQ4NCwicGF0aCI6Ii84Mzg4MjQ5Ny80MzAzMDM3MTgtNDZiMGNiZjAtNDlhZC00YjMyLWJiYWUtMDIwZjY2YjNlZTZiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDA0VDEwMjgwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThkMTBiNDFmNjhhZGE0ZWQwZjgwMmU2NjBlMmI3MDIyOTYxZTYwNDc1MDhiY2FiYTYwZGVhNTBkOWZjMTRjMGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.gEkBjXD-OUppMDeske1MIot5GEYSCztaGt4XHdR0K3Y)
> 
> My suggestion is to use: https://github.com/alexbelgium/hassio-addons
> 
> Add the Repository to the Home Assistant Add-ons (refresh page after adding) and then look for "Filebrowser", install and run.
> 
> It's like a Windows File Explorer for Home Assistant (use with caution of course).
> 
> Hope that helps ...

Thx for the hint. I was hoping, that this one does folder upload, but it only allows file selection, even if the folder upload option is shown. I'll try the ftp or samba way later at home or upload every single file with folder creation ;-l pheeeew...

Running HA in WMware on a windows machine.

vulcanoid1975 commented:
> Thx for the hint. I was hoping, that this one does folder upload, but it only allows file selection, even if the folder upload option is shown. I'll try the ftp or samba way later at home or upload every single file with folder creation ;-l pheeeew...
> 
> Running HA in WMware on a windows machine.

I have a cheap mini pc just for Home Assistant OS x86 so I have all the options ... I just uploaded the whole folder and replaced all duplicate files ...
Good luck!

Passi66 commented:
> > Thx for the hint. I was hoping, that this one does folder upload, but it only allows file selection, even if the folder upload option is shown. I'll try the ftp or samba way later at home or upload every single file with folder creation ;-l pheeeew...
> > Running HA in WMware on a windows machine.
> 
> I have a cheap mini pc just for Home Assistant OS x86 so I have all the options ... I just uploaded the whole folder and replaced all duplicate files ... Good luck!

Yes, that will be probably my next move. Away from this old 0€ thin pc to maybe a second raspi or...wait, this old one could be also an only HA-PC maybe, good idea! 
Thx!

Passi66 commented:
> > > I know to use Studio Code Server but not sure how to add the files to it? or if i replace them?
> > > ![Studio Code Server](https://private-user-images.githubusercontent.com/83882497/430303718-46b0cbf0-49ad-4b32-bbae-020f66b3ee6b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDM3NjI3ODQsIm5iZiI6MTc0Mzc2MjQ4NCwicGF0aCI6Ii84Mzg4MjQ5Ny80MzAzMDM3MTgtNDZiMGNiZjAtNDlhZC00YjMyLWJiYWUtMDIwZjY2YjNlZTZiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDA0VDEwMjgwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThkMTBiNDFmNjhhZGE0ZWQwZjgwMmU2NjBlMmI3MDIyOTYxZTYwNDc1MDhiY2FiYTYwZGVhNTBkOWZjMTRjMGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.gEkBjXD-OUppMDeske1MIot5GEYSCztaGt4XHdR0K3Y)
> > 
> > 
> > My suggestion is to use: https://github.com/alexbelgium/hassio-addons
> > Add the Repository to the Home Assistant Add-ons (refresh page after adding) and then look for "Filebrowser", install and run.
> > It's like a Windows File Explorer for Home Assistant (use with caution of course).
> > Hope that helps ...
> 
> Thx for the hint. I was hoping, that this one does folder upload, but it only allows file selection, even if the folder upload option is shown. I'll try the ftp or samba way later at home or upload every single file with folder creation ;-l pheeeew...
> 
> Running HA in WMware on a windows machine.

Update: this worked great with the Filebrowser. On my mobile phone a mess with uploading the whole thing, but on PC & Chrome with only drag and drop into the folder "/homeassistant/custom_components/)" a lovely thing! 
all my sub pages are back. Thx for the work guys!!!

j3ffrw commented:
works for me using hacs custom repository, after a few browser refresh

SirUli commented:
Works perfectly - thanks.

S0uljacker commented:
> > > Here's a new zip file up to the last commit, for whoever wants to test.
> > > [dwains_dashboard.zip](https://github.com/user-attachments/files/19591604/dwains_dashboard.zip)
> > 
> > 
> > tested it - works :)
> 
> Thank you so far all is back up and working with Dwain! thanks
> 
> I did struggle to find the directory as there is 2 folder in there! So be careful where you upload too! But ts all working!

Tested this ZIP-File with Home Assistant 2025.4.1 (on HAOS 15.1) -> Working fine at the moment! THX for saving me a lot of time!

By the way: For uploading the custom files in my HA instance I used the filebrowser HA Add-on by alexbelgium (https://github.com/alexbelgium/hassio-addons)

yurtalan92 commented:
everything works fine. But when I change the add-on card on the Dwains dashboard or change the size of anything. I have to restart the home assistant for it to be applied. It wasn't like this before. Could there be a mistake in the files? Or is this just me?

Onepamopa commented:
@yurtalan92 not just you...  @dwainscheeren can you take a look at what could have caused this?

Chrisni2 commented:
> everything works fine. But when I change the add-on card on the Dwains dashboard or change the size of anything. I have to restart the home assistant for it to be applied. It wasn't like this before. Could there be a mistake in the files? Or is this just me?

I have had this same issue since before the latest issues with More Pages, it's very time consuming to have to restart HA every time they are edited.  If it can ln be fixed, I love to know how.

Onepamopa committed:
Fix for changes not being reflected immediately

Onepamopa commented:
Here is an updated zip file:

[dwains_dashboard.zip](https://github.com/user-attachments/files/19616748/dwains_dashboard.zip)

Gmutschler2 commented:
Will the actual addon be updated in HACS or do i have to update via the workaround zip? Thanks

Onepamopa commented:
> Will the actual addon be updated in HACS or do i have to update via the workaround zip? Thanks

@dwainscheeren said he will update the addon officially in a few days. All we have to do is - test and confirm it's working normally.

etvanp commented:
Tested the zip version and confirm it's ok.

vulcanoid1975 commented:
Tested as well the new version of the file with the new Core update today (2025.4.1). 
I don't use the "More" page, but it's there and I can see it, no errors ...
Everything seems to work perfectly. 
Thank You again!

nimdaxxx commented:
> Here is an updated zip file:
> 
> [dwains_dashboard.zip](https://github.com/user-attachments/files/19616748/dwains_dashboard.zip)

I can confirm that with Core 2025.4.1 it works perfectly. Thank you!

dmelideo commented:
> > Ive noticed that its making a lot of noise in Logs, is same for You?
> 
> Yes. I also commented it out for me, but I did not want to include it in the files I uploaded.

I'm a newbie, can you clarify exactly how to overwrite the files? Thank you

vulcanoid1975 commented:
> I'm a newbie, can you clarify exactly how to overwrite the files? Thank you

Since this will be a new version available on HACS (I think) ... I would probably wait for that ... but if you want ...

Commented above:
> My suggestion is to use: https://github.com/alexbelgium/hassio-addons
> 
> Add the Repository to the Home Assistant Add-ons (refresh page after adding) and then look for "Filebrowser", install and run.
> 
> It's like a Windows File Explorer for Home Assistant (use with caution of course).

(if some of this is new to you, just follow the link and go to "Installation" on that page, there are instructions there)

Download and Unzip the file above:
> > Here is an updated zip file:
> > [dwains_dashboard.zip](https://github.com/user-attachments/files/19616748/dwains_dashboard.zip)

And use the upload function to upload to:
![image](https://github.com/user-attachments/assets/80a151c9-413e-42de-8c53-2a2ad4c46e68)

NOTE: You should upload to "custom_components" not the address in the picture, or you would be creating a folder with the same name inside the folder "dwains_dashboard".

The "arrow up" wil upload, choose "Folder" when asked and upload the unzipped folder.

It will ask you if you wish to replace files or not ... answer accordingly.

Hope that helps ...

mucmilbe5 commented:
> > Here is an updated zip file:
> > [dwains_dashboard.zip](https://github.com/user-attachments/files/19616748/dwains_dashboard.zip)
> 
> I can confirm that with Core 2025.4.1 it works perfectly. Thank you!

Same here. Tested and works like a charme. Thanks a lot!

mindtheshark commented:
Dwains Dashboard is working again; thanks for the fix/update!

Core 2025.4.1
Supervisor 2025.03.4
Operating System 15.1
Frontend 20250404.0

dmelideo commented:
Appreciate the fast responses and support.
I am still getting an error: "Config flow could not be loaded: {"message":"Invalid handler specified"}".
I will try and remove the current Dwains Dashboard instance via the renaming the "dwains-dashboard" folder method provided in the documentation. Then reinstall with the link/files you have provided.
For clarity, will the version of Dwains Dashboard v 3.7.1 be updated with the corrections noted above? If so, I will wait for version 3.x.x to be published.
Thanks to everyone who has worked to solve this issue.

Onepamopa commented:
@dmelideo can you paste the whole error from home-assistant.log?
I'm curious because I'm not getting "Config flow could not be loaded: {"message":"Invalid handler specified"}".

dmelideo commented:
@Onepamopa 

Appreciate your interest in helping out.
I may have messed up the file structure during attempts to correct my issue. In an area I'm new to within HA. Comfortable within Windows. I cleared my log, then retried the loading/installing the integration. Two log entries are pasted below. Dwains Dashboard is also not showing up in my Integration list although I do see the file structures within HA.

Logger: homeassistant.config_entries
Source: config_entries.py:3771
First occurred: 9:46:23 AM (1 occurrences)
Last logged: 9:46:23 AM

Error occurred loading flow for integration dwains_dashboard: Exception importing custom_components.dwains_dashboard.config_flow


Logger: homeassistant.loader
Source: loader.py:1249
First occurred: 9:46:23 AM (2 occurrences)
Last logged: 9:46:23 AM

Unexpected exception importing platform custom_components.dwains_dashboard.config_flow
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1249, in _load_platform
    cache[full_name] = self._import_platform(platform_name)
                       ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/loader.py", line 1281, in _import_platform
    return importlib.import_module(f"{self.pkg_path}.{platform_name}")
           ~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/util/loop.py", line 201, in protected_loop_func
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.13/importlib/__init__.py", line 88, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1310, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 1026, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/config/custom_components/dwains_dashboard/__init__.py", line 10, in <module>
    from .process_yaml import process_yaml, reload_configuration
  File "/config/custom_components/dwains_dashboard/process_yaml.py", line 69, in <module>
    loader.PythonSafeLoader.add_constructor("!include", _include_yaml)
    ^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: module 'homeassistant.util.yaml.loader' has no attribute 'PythonSafeLoader'

Onepamopa commented:
@dmelideo you didn't update from the latest zip file. Your current files are looking for homeassistant.util.yaml.loader and it's no longer there.

Kindly download the latest zip from the PR (few messages above this one). Also note that when you replace the files - you need to ***RESTART*** home assistant.

Xander-V commented:
Thanks a bunch both @Onepamopa and @dwainscheeren, higly appreciated!

Gmutschler2 commented:
So is the hacs addon now updated ? How do I make the update? Should I re-download the addon?

Onepamopa commented:
@dwainscheeren has merged the fixes into the official repository, but no new release has been created (yet).

dwainscheeren commented:
I try to release it today or tomorrow in an official release, I want to thank you all for the testing and hard work :)

Onepamopa commented:
> I try to release it today or tomorrow in an official release, I want to thank you all for the testing and hard work :)

And I say - let's keep the development going. I get you don't have much time, but still..

dwainscheeren commented:
Please try https://github.com/dwainscheeren/dwains-lovelace-dashboard/releases/tag/v3.8.0

tristanXme commented:
> Please try https://github.com/dwainscheeren/dwains-lovelace-dashboard/releases/tag/v3.8.0

Its working. Thank you.

vulcanoid1975 commented:
> > Please try https://github.com/dwainscheeren/dwains-lovelace-dashboard/releases/tag/v3.8.0
> 
> Its working. Thank you.

Same here ... Working with Core version 2025.4.1

Thank You!

ilidio-miguel commented:
For me everything is ok.

renbond commented:
mine was broken after latest HA update, finally working following this steps, took me sometime to figure out, since I had 2 different version installed, had to deleted all old dwains dashborad, now I have to assign 1000 diffrent entities/ to look how it was before  Thank you!!

Gmutschler2 commented:
When will the official version be released into hacs?

vulcanoid1975 commented:
> When will the official version be released into hacs?

https://github.com/dwainscheeren/dwains-lovelace-dashboard/issues/861#issuecomment-2794811364

Suggestion ...

Gmutschler2 commented:
> > When will the official version be released into hacs?
> 
> [#861 (comment)](https://github.com/dwainscheeren/dwains-lovelace-dashboard/issues/861#issuecomment-2794811364)
> 
> Suggestion ...

Thank you that worked.

dwainscheeren commented:
https://github.com/dwainscheeren/dwains-lovelace-dashboard/discussions/872

digibaro commented:
After applying v3.8.0 and upgrading to HA 2025.04.04 all Dwains (sub) Dashboards are blank. Only at the beginning the Dwains Dashboard top bar is visible but that will disappear if I click a (sub)dashboard. I have tested in Firefox and Edge with clearing of the browser cache.

dmelideo commented:
Had similar issues. Created more trying to fix it. Updating to ver 2025.xx along with redownloading Dwain’s v3.8.0 and it worked. I’m using latest HA:

  *   Core2025.5.0b3
  *   Supervisor2025.04.1


Thank you
Dan
________________________________
From: digibaro ***@***.***>
Sent: Monday, May 5, 2025 5:38:46 AM
To: dwainscheeren/dwains-lovelace-dashboard ***@***.***>
Cc: dmelideo ***@***.***>; Mention ***@***.***>
Subject: Re: [dwainscheeren/dwains-lovelace-dashboard] Additional fixes for PR 862 (PR #863)

[https://avatars.githubusercontent.com/u/83860720?s=20&v=4]digibaro left a comment (dwainscheeren/dwains-lovelace-dashboard#863)<https://github.com/dwainscheeren/dwains-lovelace-dashboard/pull/863#issuecomment-2850452053>

After applying v3.8.0 and upgrading to HA 2025.04.04 all Dwains (sub) Dashboards are blank. Only at the beginning the Dwains Dashboard top bar is visible but that will disappear if I click a (sub)dashboard. I have tested in Firefox and Edge with clearing of the browser cache.

—
Reply to this email directly, view it on GitHub<https://github.com/dwainscheeren/dwains-lovelace-dashboard/pull/863#issuecomment-2850452053>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/A2T4HZS6U7QQPMAYBGUXJPD244WSNAVCNFSM6AAAAABZ26WDW2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDQNJQGQ2TEMBVGM>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

dwainscheeren commented:
As you all maybe saw I'm back! :) DD 4.0 is now in beta on my Discord!
